src/Grid.ts

```diff
@@ -79,18 +85,35 @@ class Grid extends GridBase<GridProperties> {
 			role: 'grid'
 		}, [
 			w<Header>('header', {
-				columns,
 				registry,
-				sortDetails,
-				theme,
-				onSortRequest
-			}),
+				theme
+			}, <DNode[]> headers.map((child) => {
```

Comment:
Seems odd we need a cast here.  Also this logic is quite hard to follow in line and is duplicated below.  This should probably be broken into a function that is easier to grok and maintain.